frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,358 @@
+/**
+ * @jest-environment jsdom
+ */
+import { buildAccountDataForEncryption, encryptAccountData, decryptData } from '../src/utils/account-send';
+import { BigNumber } from '../src/utils/ethers';
+
+describe('buildAccountDataForEncryption Utils', () => {
+  const recipientAddress = '0x2436012a54c81f2F03e6E3D83090f3F5967bF1B5';
+  const pubKey =
+    '0x0476698beebe8ee5c74d8cc50ab84ac301ee8f10af6f28d0ffd6adf4d6d3b9b762d46ca56d3dad2ce13213a6f42278dabbb53259f2d92681ea6a0b98197a719be3';
+
+  it('invalid address', () => {
+    let err;
+    try {
+      buildAccountDataForEncryption({
+        recipientAddress: 'adfaklsfjl',
+        advancedMode: true,
+        pubKey: pubKey,
+        usePublicKeyChecked: true,
+      });
+    } catch (e) {
+      err = e;
+    }
+    expect((err as Error)?.message).toBe('Invalid recipient address');
+  });
+
+  it('invalid pubkey prefix', () => {
+    let err;
+    try {
+      buildAccountDataForEncryption({
+        recipientAddress,
+        advancedMode: true,
+        pubKey: '0x039dad8ddb0bd43093435ce4',
+        usePublicKeyChecked: true,
+      });
+    } catch (e) {
+      err = e;
+    }
+    expect((err as Error)?.message).toBe('Invalid public key prefix');
+  });
+
+  it('invalid pubkey', () => {
+    let err;
+    try {
+      buildAccountDataForEncryption({
+        recipientAddress,
+        advancedMode: true,
+        pubKey: '0x049dad8ddb0bd43093435ce4',
+        usePublicKeyChecked: true,
+      });
+    } catch (e) {
+      err = e;
+    }
+    expect((err as Error)?.message).toBe('Invalid public or private key');
+  });
+
+  it('Data is correct', () => {
+    const data = buildAccountDataForEncryption({
+      recipientAddress,
+      advancedMode: true,
+      pubKey: pubKey,
+      usePublicKeyChecked: true,
+    });
+    expect(data === BigNumber.from(`${recipientAddress}11${pubKey.slice(4)}`));
```

Comment:
I've already fix this locally, but @alexkeating just a heads up this test always passes because the `expect` statement is never finished. If you changed this to the below, you'd see the test fail
```suggestion
    expect(data === BigNumber.from(`${recipientAddress}11${pubKey.slice(4)}`)).toEqual(true);
```

I've updated this to also compare the hex string representations, since I always forget how comparing objects with `===` works so working with strings feels safer